custom_connectors/oauth2/google_analytics_connector.rb

```diff
@@ -0,0 +1,877 @@
+        },
+        {
+          name: "dimension_filter_fields",
+          control_type: "multiselect",
+          pick_list: "dimensions",
+          pick_list_params: { account_id: "account", property_id: "property" },
+          delimiter: "\n",
+          hint: "Select a property to view list of dimensions to filter by"
+        },
+		    {
+          name: "metric_filter_fields",
+          control_type: "multiselect",
+          pick_list: "metrics",
+          pick_list_params: { account_id: "account", property_id: "property" },
+          delimiter: "\n",
+          hint: "Select a property to view list of metrics to filter by"
+        }
+      ],
+
+      input_fields: lambda do |object_definitions|
+        object_definitions["filters"]
+      end,
+
+      output_fields: lambda do |object_definitions|
+        object_definitions["report_output"]
+      end,
+
+      execute: lambda do |_connection, input|
+        dimensions = input["dimensions"].split("\n")
+        metrics = input["metrics"].split("\n")
+
+        dimension_filter_clauses =
+          if(input["dimension_filters"].blank?)
+						[]
+					else
+            [{
+              "operator" => "AND",
+              "filters"	=>	[
+                              (input["dimension_filters"] || {}).
+                                map do |k, v|
+                                  if( v.present? )
+                                    {
+                                      "dimensionName" => "ga:#{k}",
+                                      "operator"=>"EXACT",
+                                      "expressions"=> ["#{v}"]
+                                    }
+                                  else
+                                    nil
+                                  end
+                                end
+                            ]
+            }]
+					end
+
+        metric_filter_clauses =
+          if(input["metric_filters"].blank?)
+						[]
+					else
+            [{
+              "operator" => "AND",
+              "filters"	=>	[
+                              (input["metric_filters"] || {}).
+                                map do |k, v|
+                                  if( v.present? )
+                                    {
+                                      "metricName" => "ga:#{k}",
+                                      "operator"=>"EQUAL",
+                                      "comparisonValue"=> "#{v}"
+                                    }
+                                  else
+                                    nil
+                                  end
+                                end
+                            ]
+              }]
+					end
+
+        payload = {
+          "reportRequests" => [
+            {
+              "viewId" => input["view"],
+              "dateRanges" => [
+                {
+                  "startDate": (Time.now - 7.days).to_date.iso8601,
```

Comment:
we can use input fields (`Start date` & `End date` )to set the date-range for reports, like:
```"startDate": input["start_date"].to_date.iso8601,
"endDate": input["end_date"].to_date.iso8601,
```